spring-integration-gcp/src/main/java/org/springframework/integration/gcp/storage/GcsSession.java

```diff
@@ -189,6 +191,10 @@ public Object getClientInstance() {
 		// Assumes paths of the form bucket/folder/blob
 		Assert.hasText(path, "Path can't be empty.");
 
-		return path.split("/", 2);
+		return path.split(this.remoteFileSeparator, 2);
```

Comment:
I wonder if this is something what is allowed by Google Storage. I mean AWS S3 allows only `/` as a virtual directory separator.

WDYT?

Reply:
That's a good point.

Although the paths we're getting here can come from e.g., `AbstractRemoteFileStreamingMessageSource` which uses a `remoteFileSeparator` that can be set to something else.
But it's probably safe to assume that this wouldn't be set to anything else other than  `/`, so let's rollback this bit for now.

Reply:
We can override `setRemoteFileSeparator()` there with the `UnsupportedOperationException`.

Out of topic: GH inhibits for me here, especially when I comment, like there is some hard JavaScript.
Don't you know what that might be?

Reply:
Do you mean GH is hiding the comments?
If so, that may be because in my last commits I changed the lines being commented.

Unless you mean something else?

Reply:
Yeah.. Sure. I mean it is very slow and doesn't let me to comment properly like it is reloading page on each my key press